### PR TITLE
fix: validate that a flag key is valid UTF-8 & implemented fuzzing tests

### LINKED ISSUE #127: [TESTS] Introduce fuzzing tests
### Requirements

Fuzzing is a [repository requirement](https://github.com/open-feature/.github/blob/main/CONTRIBUTING.md#repository-requirements).

Go supports fuzzing natively since 1.18, `go-sdk` is still using version 1.17. Either upgrade to 1.18 or use a third-party library.

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

- Fuzzing tests.
- Validate that a flag key is valid UTF-8 (issue arose from fuzzing 🎉 )

### Related Issues

Fixes #127

### CODE REVIEW

**Comment by skyerus** [comment]
Will rebase once 1.18 is merged: https://github.com/open-feature/go-sdk/pull/140

**Review by beeme1mr — APPROVED** [review]

File: pkg/openfeature/client.go [hunk 1]
```diff
@@ -563,6 +564,18 @@ func (c *Client) evaluate(
...
+		Value: defaultValue,
+		EvaluationDetails: EvaluationDetails{
+			FlagKey:  flag,
+			FlagType: flagType,
+		},
+	}
+
+	if !utf8.Valid([]byte(flag)) {
+		return evalDetails, NewParseErrorResolutionError("flag key is not a UTF-8 encoded string")
+	}
```
Comment (toddbaert) [inline pkg/openfeature/client.go:577]: Besides flag key, is there any other places we should validate this?

Is it worth doing it for context keys and values, for instance (that could incur some performance penalties... Just a thought!

Comment (skyerus) [reply pkg/openfeature/client.go:577]: Perhaps there is value to that, should we create a separate issue to raise the question?

**Comment by Kavindu-Dodan** [comment]
Overall looks good. One suggestion, let's add few more fuzz seeds. For example, flag keys with mix cases, include numbers (A-z0-9) plus special characters if needed. Plus the same for default variant to match the type.

File: pkg/openfeature/client.go (hunk 1 above)
Comment (skyerus) [reply pkg/openfeature/client.go:577]: https://github.com/open-feature/go-sdk/issues/147

**Comment by skyerus** [comment]
> Overall looks good. One suggestion, let's add few more fuzz seeds. For example, flag keys with mix cases, include numbers (A-z0-9) plus special characters if needed. Plus the same for default variant to match the type.

Thanks! Good suggestion, done :)

**Review by Kavindu-Dodan — APPROVED** [review]

**Pull request merged**